test/test_response.py

```diff
@@ -377,6 +377,59 @@ def test_empty_stream(self):
 
         self.assertRaises(StopIteration, next, stream)
 
+    def test_length_no_header(self):
+        fp = BytesIO(b'12345')
+        resp = HTTPResponse(fp, preload_content=False)
+        assert resp.length is None
+
+    def test_length_w_valid_header(self):
+        headers = {"content-length": "5"}
+        fp = BytesIO(b'12345')
+
+        resp = HTTPResponse(fp, headers=headers, preload_content=False)
+        assert resp.length == 5
+
+    def test_length_w_bad_header(self):
+        garbage = {'content-length': 'foo'}
+        fp = BytesIO(b'12345')
+
+        resp = HTTPResponse(fp, headers=garbage, preload_content=False)
+        assert resp.length is None
+
+        garbage['content-length'] = "-10"
+        resp = HTTPResponse(fp, headers=garbage, preload_content=False)
+        assert resp.length is None
+
+    def test_length_when_chunked(self):
+        headers = {'content-length': '5',
+                   'transfer-encoding': 'chunked'}
+        fp = BytesIO(b'12345')
+
+        resp = HTTPResponse(fp, headers=headers, preload_content=False)
+        assert resp.length is None
+
+    def test_length_after_read(self):
+        headers = {"content-length": "5"}
+
+        # Test no defined length
+        fp = BytesIO(b'12345')
+        resp = HTTPResponse(fp, preload_content=False)
+        resp.read()
+        assert resp.length is None
+
+        # Test our update from content-length
+        fp = BytesIO(b'12345')
+        resp = HTTPResponse(fp, headers=headers, preload_content=False)
+        resp.read()
+        assert resp.length == 0
+
+        # Test partial read
+        fp = BytesIO(b'12345')
+        resp = HTTPResponse(fp, headers=headers, preload_content=False)
+        data = resp.stream(2)
+        next(data)
+        assert resp.length == 3
```

Comment:
I'm fine with renaming it, I was just trying to keep this in sync with `httplib.HTTPResponse.length` so you can call `length` on either object and get back the same result.